app/javascript/components/reconfigure-vm-form/index.jsx

```diff
@@ -0,0 +1,270 @@
+        title: formData[type].title,
+        type,
+        action,
+        className: formData[type].className,
+      },
+    });
+  };
+
+  /** Function to handle the button clicks of the datatable. */
+  const setButtonClick = (item) => {
+    if (item && item.callbackAction) {
+      if (item.callbackAction === 'editNetwork') {
+        setRenderForm(item, 'network', 'edit');
+      } else if (item.callbackAction === 'connectDrives') {
+        setRenderForm(item, 'drive', 'edit');
+      } else if (item.callbackAction === 'resizeDisk') {
+        setRenderForm(item, 'disk', 'resize');
+      } else {
+        setButtonActions(item, data, setData, roles);
+      }
+    }
+  };
+
+  /** Function to execute the click events of from the datables.
+  * Like - row's click event  for resize and delete button click event.
+  */
+  const onCellClick = (selectedRow, cellType) => {
+    if (cellType === CellAction.buttonCallback) {
+      setButtonClick(selectedRow);
+    }
+    return true;
+  };
+
+  let diskTable = '';
+  let networkTable = '';
+  let cdromTable = '';
+  let clearOnUnmount = true;
+  if (data.form.type === 'reconfigure') {
+    clearOnUnmount = false;
+  }
+  if (data.form.type === 'reconfigure' && recordId.length === 1) {
+    if (roles.allowDiskChange) {
+      diskTable = (
+        <DiskTable
+          headers={dataTableHeaders('disk', roles)}
+          rows={setDiskData(data.dataTable.disks, roles, data, setData)}
+          onDiskCellClick={onCellClick}
+          buttonClick={setRenderForm}
+        />
+      );
+    }
+
+    if (roles.allowNetworkChange) {
+      networkTable = (
+        <NetworkTable
+          headers={dataTableHeaders('network', roles)}
+          rows={setNetworkData(data.dataTable.networkAdapters, roles, data, setData)}
+          onNetworkCellClick={onCellClick}
+          buttonClick={setRenderForm}
+        />
+      );
+    }
+
+    if (data.dataTable.drives.length > 0 && roles.allowCdromsChange) {
+      cdromTable = (
+        <CdDriveTable
+          headers={dataTableHeaders('drive', roles)}
+          rows={setDrivesData(data.dataTable.drives, data)}
+          onDriveCellClick={onCellClick}
+        />
+      );
+    }
+  }
+  let canReset = true;
+  let hasCustomTemplate = true;
+  if (data.form.type !== 'reconfigure') {
+    canReset = false;
+    hasCustomTemplate = false;
+  }
+
+  const renderMiqForm = () => (hasCustomTemplate ? (
+    <MiqFormRenderer
+      schema={createSchema(recordId, data, setData, roles, options, memory)}
+      initialValues={data.initialValues}
+      canReset={canReset}
+      onSubmit={onSubmit}
+      onCancel={() => onCancel(data)}
+      validatorMapper={customValidatorMapper}
+      hasCustomTemplate={hasCustomTemplate}
+      buttonsLabels={{
+        submitLabel: __('Submit'),
+      }}
+      clearOnUnmount={clearOnUnmount}
+      canSubmit={getFormSubmitStatus(data)}
+    >
+      {diskTable}
+      {networkTable}
+      {cdromTable}
```

Comment:
can you try this with the `componentMapper` and include them in schema? (if possible)